File: SocialBookAppDomain/Book.cs

namespace SocialBookAppDomain;

public class Book
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public long ISBNNumber { get; set; }
    public List<Author> Authors { get; set; } = new List<Author>();
    public DateTime Published { get; set; }
    public Genre Genre { get; set; }
    public string Language { get; set; } = string.Empty;
    public BookFormat Format { get; set; }
    public bool IsChildFriendly { get; set; }
    public int Pages { get; set; }
    public int Chapters { get; set; }
    public string Blurb { get; set; } = string.Empty;
    public List<Review> Reviews { get; set; } = new List<Review>();
}

File: SocialBookAppVue/src/types/book.ts
import type { Author } from './author.ts'
import type { Review } from './review.ts'

export interface Book {
  id?: number
  title: string
  isbnNumber?: number
  authors: Author[]
  published: string
  genre: string
  language: string
  format: string
  isChildFriendly: boolean
  pages?: number
  chapters?: number
  blurb: string
  reviews: Review[]
}
